### ISSUE TITLE
Cilium allocates random allocation CIDR on startup in pure multi-pool mode

### ISSUE THREAD

**Issue by CallMeFoxie** [cilium/cilium#28632]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

It seems that Cilium allocates a random IP on startup with `Using autogenerated IPv4 allocation range` message. We do not specify any CIDR locally as we use multi-pool mode.

This generated IP is now also announced via BGP, eventually going to hit our actual ranges and make a havoc on the network.

It would be nice if I could disable this (for me) random IP.

Thanks
Ashley

### Cilium Version

1.15.0 branch

### Kernel Version

5.15

### Kubernetes Version

1.27.1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by CallMeFoxie** [comment]
Meaning I get something like
```
# cilium bgp routes advertised ipv4 unicast peer 10.255.20.111    
VRouter      Prefix           NextHop         Age        Attrs
4294967294   10.187.0.0/16    10.244.21.187   16h49m3s   [{Origin: i} {AsPath: } {Nexthop: 10.244.21.187} {LocalPref: 100}]   
4294967294   10.89.4.144/28   10.244.21.187   16h49m3s   [{Origin: i} {AsPath: } {Nexthop: 10.244.21.187} {LocalPref: 100}]   
4294967294   10.89.4.176/28   10.244.21.187   16h49m3s   [{Origin: i} {AsPath: } {Nexthop: 10.244.21.187} {LocalPref: 100}] 
```
where (obviously) the one is not something I want and very close to actual subnets...

**Comment by CallMeFoxie** [comment]
For the record we use this argument
```
level=info msg="  --auto-create-cilium-pod-ip-pools='default=ipv4-cidrs:10.89.4.0/24;ipv4-mask-size:28;ipv6-cidrs:2a02:598:64:8a00::2080:0/105;ipv6-mask-size:120'" subsys=daemon
```

**Comment by CallMeFoxie** [comment]
It seems that setting `exportPodCIDR` to `false` stops it at least from being announced on BGP, that's better than nothing :)

**Comment by gandro** [comment]
This is related to https://github.com/cilium/cilium/issues/27018 

As a potential mitigation, we could teach the BGP code to ignore the local alloc CIDR in any IPAM mode other than `kubernetes` or `cluster-pool`, as all other IPAM modes (multi-pool, eni, etc) do not have a per-node podCIDR.

**Comment by YutaroHayakawa** [comment]
In BGP, in the case of the multi-pool IPAM, users should rely on `podIPPoolSelector` and shouldn't rely on `exportPodCIDR`. So, disabling `exportPodCIDR` is the right solution in this particular case, but I learned `exportPodCIDR` is confusing though.

However, as you highlighted, we might have a case in which we advertise a local alloc CIDR when we use `kubernetes` or `cluster-pool` IPAM, and PodCIDR is not assigned. Which to my knowledge, doesn't make sense most of the time.

I think we can tackle this issue if we can distinguish auto-generated PodCIDR and PodCIDR assigned properly by IPAM. Do we have such a way? @gandro

**Comment by gandro** [comment]
> I think we can tackle this issue if we can distinguish auto-generated PodCIDR and PodCIDR assigned properly by IPAM. Do we have such a way? @gandro

We have no such way, besides checking the running IPAM mode. My long-term hope is that we can just remove the auto-generated pod CIDR logic (or make it opt-in) (c.f. https://github.com/cilium/cilium/issues/27018). The result should be that if there is a pod CIDR, then it's a real one.

**Comment by YutaroHayakawa** [comment]
> We have no such way, besides checking the running IPAM mode. My long-term hope is that we can just remove the auto-generated pod CIDR logic (or make it opt-in) (c.f. https://github.com/cilium/cilium/issues/27018).

Yep, that makes sense to me.

**Comment by CallMeFoxie** [comment]
I dare you to close this, my dear bot! :-)

**Comment by bewing** [comment]
Even if the random allocated CIDR isn't advertised via BGP, I still see it installed into the routing table for the node (pointing at `cilium_host`) when configured in native routing / eBPF mode.

At the very worst, it's creating random blackholes if pods are trying to access off-cluster resources in that range (assuming on-prem and we're using that RFC1918 space elsewhere in our infrastructure)

At the very least, it's adding confusion

**Comment by bewing** [comment]
Testing shows that the route doesn't affect pods (because eBPF forwards it out), but the node itself considers it a blackhole.

LMK if I should file a separate issue for this.

**Comment by YutaroHayakawa** [comment]
@bewing Yep, please file a separate issue if there's no existing one. Thanks!
